infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -174,13 +173,22 @@ private PollReport pollRound(long epoch,
                 // Timeout is increased anytime a failure is encountered.
                 // Failure includes both, unresponsive and outOfPhaseEpoch nodes.
                 if (failuresDetected) {
+                    final long iterationElapsedTime = TimeUnit.MILLISECONDS.convert(
+                            System.nanoTime() - iterationStartTime, TimeUnit.NANOSECONDS);
+
+                    // Increase the inter iteration sleep time if the period time increases
+                    interIterationSleepTime = Math.max(interIterationInterval,
+                                                       period - iterationElapsedTime);
```

Comment:
If you can only sleep in milliseconds and you ceil to 1ms, whats the point of tracking nanoseconds and convertin ns->ms ?

Reply:
nanoTime provides relative time rather than wall clock and is a better choice for micro benchmarking. its better accuracy is the secondary benefit.